dataretrieval/nwis.py

```diff
@@ -845,12 +845,24 @@ def _set_metadata(response, **parameters):
             unpacked dictionary of the parameters supplied in the request
     """
     md = set_md(response)
-    site_aliases = ['sites', 'site_no']
-    for alias in site_aliases:
-        if alias in parameters:
-            md.site_info = lambda: what_sites(sites=parameters[alias])
-            break
+    # site_no is preferred over sites to set site_info if both are present,
```

Comment:
This part feels a little kludgy, but I'd have to think more about how we improve it.